drizzlepac/haputils/analyze.py

```diff
@@ -471,3 +483,163 @@ def generate_msg(filename, msg, key, value):
         log.warning('Dataset cannot be aligned.')
     else:
         log.warning('Dataset can be aligned, but the result may be compromised.')
+
+
+# -----------------------------------------------------------------------------
+#  Line detection functions
+# -----------------------------------------------------------------------------
+
+def verify_guiding(filename, min_length=33):
+    """ Verify whether or not the input image was affected by guiding problems.
+
+    This algorithm evaluates the data from (SCI,1) to try to determine whether
+    the image was affected by guiding problems which mimic SCAN mode or GRISM data
+    with the trails in an arbitrary angle across the image.
+
+    Parameters
+    ==========
+    filename : str
+        Name of image to be evaluated
+
+    min_length : int, optional
+        Minimum length of trails (in pixels) to be detected in image.
+
+    Returns
+    ========
+    bad_guiding : bool
+        Boolean specifying whether or not the image was detected as
+        being affected by guiding problems.  Value is True if image
+        was affected.
+    """
+    log.info(f"Verifying that {filename} was not affected by guiding problems.")
+
+    hdu = fits.open(filename)
+    data = hdu[("SCI", 1)].data.copy()
+    scale_data = hdu[("SCI",1)].header["bunit"].endswith('/S')
+    data = np.nan_to_num(data, nan=0.0)  # just to be careful
+    if scale_data:
+        # Photutils works best in units of DN
+        scale_hdr = hdu[0].header if 'exptime' in hdu[0].header else hdu[1].header
+        scale_val = scale_hdr['exptime']
+        data *= scale_val
+    bkg_stats = sigma_clipped_stats(data, maxiters=2)
+    bkg_limit = bkg_stats[1] + bkg_stats[2]
```

Comment:
Please add comments for explanation. It looks like the bkg_limit is the combination of the median and stddev.

Reply:
Fixed in 9acdf37.